sdk/python/test_e2e/test_e2e_train_api.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This shell script is used to run Katib Experiment.
+# Input parameter - path to Experiment yaml.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+cd "$(dirname "$0")"
+
+echo "Training Operator deployments"
+kubectl -n kubeflow get deploy
+echo "Training Operator services"
+kubectl -n kubeflow get svc
+echo "Training Operator pods"
+kubectl -n kubeflow get pod
+echo "Training Operator persistent volume claims"
+kubectl get pvc -n kubeflow
+echo "Available CRDs"
+kubectl get crd
```

Comment:
Also is this shell script needed? We can run the python file directly.

Reply:
@deepanker13 It's not strictly necessary to have the shell script, as we could run the Python file directly. I referenced to the e2e test in the tune API [here](https://github.com/kubeflow/katib/blob/a524f33830e02189476efaf6d9045cbd2ce605f0/test/e2e/v1beta1/scripts/gh-actions/run-e2e-tune-api.sh#L32). I think the shell script mainly helps users gather more information about the configuration.

Would you suggest deleting the following lines and running the Python file directly instead?
``` shell
echo "Training Operator deployments"
kubectl -n kubeflow get deploy
echo "Training Operator services"
kubectl -n kubeflow get svc
echo "Training Operator pods"
kubectl -n kubeflow get pod
echo "Training Operator persistent volume claims"
kubectl get pvc -n kubeflow
echo "Available CRDs"
kubectl get crd
```

Reply:
@helenxie-bit I think for Training Operator we have a little bit different E2Es tests from Katib.
We define them here: https://github.com/kubeflow/training-operator/tree/9cc5429c22c589c21120f1e77b14ecc8079470a8/sdk/python/test/e2e
And we run them via `pytest` here: https://github.com/kubeflow/training-operator/blob/9cc5429c22c589c21120f1e77b14ecc8079470a8/.github/workflows/integration-tests.yaml#L98.

Maybe we should just add another tests function in the test_e2e_pytorchjob.py: https://github.com/kubeflow/training-operator/blob/9cc5429c22c589c21120f1e77b14ecc8079470a8/sdk/python/test/e2e/test_e2e_pytorchjob.py#L219

Reply:
@andreyvelich Thank you for your suggestions! I will update accordingly.